### PR TITLE
Os 130 : DB sharding 

### PR DESCRIPTION
**Pull request by pritha-tarento**
Indroducing shards:
Create a factory for DatabaseProvider. Based on provider constant the database provider instantiated.
Added models for holding the shards info as given in application yaml. Initialize the shards info components.
Added a default shard (as shard1) in application configutation.
Tested by postman

### CODE REVIEW

File: java/registry/src/main/java/io/opensaber/registry/config/GenericConfiguration.java [hunk 1]
```diff
@@ -77,6 +74,8 @@
 
 	private static Logger logger = LoggerFactory.getLogger(GenericConfiguration.class);
 
+/*	@Autowired
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/config/GenericConfiguration.java]: Remove

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfo.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	public String getPassword() {
+		return password;
+	}
+	public void setPassword(String password) {
+		this.password = password;
+	}
+	
+	@Override
+    public String toString() {
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfo.java]: is this being used?

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * @author Pritha Chattopadhyay
+ * Auto populate/bind the field values from yaml properties.
+ *
+ */
+@Component
+@ConfigurationProperties(prefix = "database")
+public class DBConnectionInfoMgr {
+	
+	/**
+	 * get the provider property value from properties or yaml file
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: I didn't mean this to be instruction; I meant we must explain/reason out why only one provider is there, but there are several connections. Communicate as you do to me orally.
"We expect only one type of database as the target as of today."
`private String provider;`
"Each DBConnectionInfo can be treated as a shard connection information"
`private List<DBConnectionInfo> connectionInfo = new ArrayList<>();`

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+	public List<DBConnectionInfo> getConnectionInfo() {
+		return connectionInfo;
+	}
+
+	public void setConnectionInfo(List<DBConnectionInfo> connectionInfo) {
+		this.connectionInfo = connectionInfo;
+	}
+
+	//TODO: add this as a map, for optimized search.
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: No TODOs please

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	/**
+	 * To provide a connection info on based of a shard identifier(name)
+	 * @param name
+	 * @return
+	 */
+	public DBConnectionInfo getDatabaseConnection(String name){
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: you can qualify the name as "shardInstance" or "shardName" or "shardId". The getDatabaseConnection is also misleading. It must be getDBConnectionInfo(shardId)

File: java/registry/src/test/java/io/opensaber/registry/RegistryTestSuite.java [hunk 6]
```diff
@@ -13,8 +13,8 @@
 import junit.framework.Test;
 import junit.framework.TestSuite;
 
-@SuiteClasses({ RegistryDaoImplTest.class, RegistryServiceImplTest.class, EncryptionDaoImplTest.class,
-		EncryptionServiceImplTest.class, SearchServiceImplTest.class, SearchDaoImplTest.class })
+@SuiteClasses({ RegistryDaoImplTest.class,RegistryServiceImplTest.class, EncryptionDaoImplTest.class,
```
Comment (indrajra) [inline java/registry/src/test/java/io/opensaber/registry/RegistryTestSuite.java]: there must be space after a comma. I expect this file to not show up in the review request.

Comment (pritha-tarento) [reply java/registry/src/test/java/io/opensaber/registry/RegistryTestSuite.java]: Done.

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java (hunk 5 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: done

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java (hunk 4 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: Done

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java (hunk 3 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: Done

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfo.java (hunk 2 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfo.java]: NOt used. So removed

File: java/registry/src/main/java/io/opensaber/registry/config/GenericConfiguration.java (hunk 1 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/config/GenericConfiguration.java]: Done

File: java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+		return connectionInfo;
+	}
+
+	public void setConnectionInfo(List<DBConnectionInfo> connectionInfo) {
+		this.connectionInfo = connectionInfo;
+	}
+
+	/**
+	 * To provide a connection info on based of a shard identifier(name)
+	 * @param name
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/model/DBConnectionInfoMgr.java]: Done. Thanks for pointing out.

**Pull request merged**